cli/src/torii/routing.rs

```diff
@@ -119,12 +119,29 @@ pub(crate) async fn handle_queries(
     wsv: Arc<WorldStateView>,
     query_judge: QueryJudgeArc,
     pagination: Pagination,
+    sorting: Sorting,
     request: VerifiedQueryRequest,
 ) -> Result<Scale<VersionedPaginatedQueryResult>> {
     let (valid_request, filter) = request.validate(&wsv, query_judge.as_ref())?;
     let original_result = valid_request.execute(&wsv)?;
     let result = filter.filter(original_result);
-    let (total, result) = if let Value::Vec(value) = result {
+    let (total, result) = if let Value::Vec(mut value) = result {
```

Comment:
This isn't quite general-enough to work in all cases. I'd recommend using the `HasMetadata` trait and relying on the fact that most if not all things returned by queries have metadata.

Reply:
I realized that `Asset` doesn't implement `HasMetadata`. Probably because an asset does not always have a metadata. We have to rethink `HasMetadata` so that it returns `Option<Metadata>`:
```rust
trait HasMetadata {
  fn metadata(&self) -> Option<Metadata>;
}
```